dora/core/common/src/main/java/alluxio/membership/EtcdMembershipManager.java

```diff
@@ -93,16 +94,25 @@ public void join(WorkerInfo workerInfo) throws IOException {
     if (existingEntityBytes != null) {
       // It's not me, or not the same me.
       if (!Arrays.equals(existingEntityBytes, serializedEntity)) {
-        // In k8s this might be bcos worker pod restarting with the same worker identity
-        // but certain fields such as hostname has been changed. Register to ring path anyway.
-        WorkerServiceEntity existingEntity = new WorkerServiceEntity();
-        existingEntity.deserialize(existingEntityBytes);
-        LOG.warn("Same worker entity found bearing same workerid:{},"
-            + "existing WorkerServiceEntity to be overwritten:{},"
-            + "maybe benign if pod restart in k8s env or same worker"
-            + " scheduled to restart on another machine in baremetal env.",
-            workerInfo.getIdentity().toString(), existingEntity);
-        mAlluxioEtcdClient.createForPath(pathOnRing, Optional.of(serializedEntity));
+        if (mConf.isSet(PropertyKey.WORKER_IN_K8S_ENV)
+            && mConf.getBoolean(PropertyKey.WORKER_IN_K8S_ENV)) {
+          // In k8s this might be bcos worker pod restarting with the same worker identity
+          // but certain fields such as hostname has been changed. Register to ring path anyway.
+          WorkerServiceEntity existingEntity = new WorkerServiceEntity();
+          existingEntity.deserialize(existingEntityBytes);
+          LOG.warn("Same worker entity found bearing same workerid:{},"
+                  + "existing WorkerServiceEntity to be overwritten:{},"
+                  + "maybe benign if pod restart in k8s env or same worker"
+                  + " scheduled to restart on another machine in baremetal env.",
+              workerInfo.getIdentity().toString(), existingEntity);
+          mAlluxioEtcdClient.createForPath(pathOnRing, Optional.of(serializedEntity));
+        } else {
+          //
+          throw new AlreadyExistsException(
+              "Some other member with same id registered on the ring, bail."
+              + "Different workers can't assume same worker identity in non-k8s env."
+                  + "Clean local worker identity settings to continue.");
```

Comment:
include in the exception message the conflicting worker's addresses and the ID

Reply:
done